### PR TITLE
Test/e2e/timer

### PR DESCRIPTION
**Pull request by francelu**
# Add End-to-End Test for Timer

## Description  
This PR introduces a comprehensive end-to-end test for the timer functionality in the `PeriodPals` Android application. The test navigates through the sign-up, profile creation, and timer actions to ensure the entire flow works as expected.  

It closes issue #301.

#### New End-to-End Test  
- **Added** `EndToEndTimer` class in `EndToEndTimer.kt` to test the complete user flow from sign-in to using the timer.  
- Utilized `composeAndroidTestRule` for setting up the test environment and interacting with UI components.  
- Implemented navigation and interaction steps to simulate user actions such as signing up, creating a profile, and using the timer.  
- Included assertions to verify that each screen and action displays the expected results, ensuring the integrity of the user flow.  
- Delete account through settings UI.

#### Added  
- `app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndTimer.kt`  

## Testing  
- Tested user flow from sign-in, through profile screen, bottom navigation menu, to timer usage using assertions to confirm screen and action validity. Deleted the account through settings UI.

### CODE REVIEW

File: app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndSignUp.kt [hunk 1]
```diff
@@ -64,8 +65,7 @@ class EndToEndSignUp : TestCase() {
    * logged in and log them out if they are. Create a new account and its profile for the test.
    */
   @Before
-  fun setUp() {
-
+  fun setUp() = runBlocking {
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndSignUp.kt:68]: Thanks for the fix!

File: app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndTimer.kt [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    }
+
+    step("User starts the timer, stops it, starts it again, and resets it") {
+      composeTestRule.waitForIdle()
+      composeTestRule
+          .onNodeWithTag(TimerScreen.START_BUTTON)
+          .performScrollTo()
+          .assertIsDisplayed()
+          .performClick()
+      SystemClock.setCurrentTimeMillis(System.currentTimeMillis() + 6L * HOUR + 30L * MINUTE)
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndTimer.kt:197]: Good thinking setting the system clock

**Review by charliemangano — APPROVED** [review]
As always, good job on the end-to-ends! LGTM!

**Review by Harrish92 — APPROVED** [review]
## Summary
Good job on your work. It was not easy to make it work, but you did it
## Important
#### Code Quality
- The code is understandable and the tests are described at every steps. 
## Testing
- testing is complete. Good job.
## Steps before PR approved
- LGTM

**Pull request merged**